### PR TITLE
chore: add SENDGRID_API_KEY to backend secrets

### PR DESCRIPTION
**Pull request by christellevs**
Related to  / https://github.com/ubio/squad-nodescript/issues/667

### CODE REVIEW

File: secrets/apps/production/backend-secrets.yaml [hunk 1]
```diff
@@ -23,8 +24,8 @@ sops:
     azure_kv: []
     hc_vault: []
     age: []
-    lastmodified: "2023-10-26T13:50:16Z"
-    mac: ENC[AES256_GCM,data:ssrezlSb6MbZQeSpfQ7LZS5KGSWBvqKfvSj6CyUjriTj//al9AKfuX3peCiA1k5yIR+rFec7ZCbhFCyz++cKiNEkYAYImNkIzsBBX/y/sxflV4KuuKe3wElBbNjAnYuYdCIxOnNwVTNuADpTf8WYrWtUIbTtTD2L0AQ+YL2H58U=,iv:iDWAat4aAkwOkE4yqUmaOWK+XZxKRJ7i6JfCJmGl2u4=,tag:2w1AD0PbouN3Y8Ig/GVzNQ==,type:str]
+    lastmodified: "2024-02-15T17:13:44Z"
+    mac: ENC[AES256_GCM,data:JPMQAjOGagmVXY/R3wY1QKtJxC/Y9bGZHBggechDjtjX5mOLDlK3xlFqKzeGVgHef1usS9ABlNt8WNn+wVqF+4S2B7eBoxdHMQWmaqU07EmPEr/gW2dJRsloiDjdoCJDoS8KnFEe+x9SMZz4Fdk7Murto2hXkmFG7UW4M439PgA=,iv:fG881fPnLNW44JzzbMY0yhloKbYieXRdZIqIJqyBgWM=,tag:LMHeT1Bj+LvfpMiBs5Wgwg==,type:str]
```
Comment (dani-beltran) [inline secrets/apps/production/backend-secrets.yaml:28]: I have no idea, what is this `mac` secret for ?

**Review by dani-beltran — APPROVED** [review]
Looks good 🍏 
I left a question, but is for me to learn.

File: secrets/apps/production/backend-secrets.yaml (hunk 1 above)
Comment (inca) [reply secrets/apps/production/backend-secrets.yaml:28]: These are generated by `sops`. If you haven't seen/used those before, I suggest taking a brief look here https://github.com/ubio/infrastructure/blob/main/README.md and here https://github.com/ubio/engineering/blob/main/docs/guides/deploying-apps.md#secrets

**Review by inca — APPROVED** [review]
Btw don't wait for approval of things like these, I'm happy if you leave those for post-reviews

**Pull request merged**